pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/operator/MailboxReceiveOperator.java

```diff
@@ -46,41 +49,56 @@ public String toExplainString() {
 
   @Override
   protected TransferableBlock getNextBlock() {
-    if (_errorBlock != null) {
-      return _errorBlock;
-    }
-    if (System.currentTimeMillis() > _context.getDeadlineMs()) {
-      _errorBlock = TransferableBlockUtils.getErrorTransferableBlock(QueryException.EXECUTION_TIMEOUT_ERROR);
-      return _errorBlock;
-    }
+    while (!_mailboxes.isEmpty()) {
+      if (_errorBlock != null) {
+        return _errorBlock;
+      }
+      if (System.currentTimeMillis() > _context.getDeadlineMs()) {
+        _errorBlock = TransferableBlockUtils.getErrorTransferableBlock(QueryException.EXECUTION_TIMEOUT_ERROR);
+        return _errorBlock;
+      }
 
-    // Poll from every mailbox in round-robin fashion:
-    // - Return the first content block
-    // - If no content block found but there are mailboxes not finished, return no-op block
-    // - If all content blocks are already returned, return end-of-stream block
-    int numMailboxes = _mailboxes.size();
-    for (int i = 0; i < numMailboxes; i++) {
-      ReceivingMailbox mailbox = _mailboxes.remove();
-      TransferableBlock block = mailbox.poll();
+      // Poll from every mailbox in round-robin fashion:
+      // - Return the first content block
+      // - If no content block found but there are mailboxes not finished, return no-op block
```

Comment:
is this still true?

Reply:
this is no longer true only for the last sentence
- if no content block found. it will call opChainContext.yield to park the current thread until notify(opChain) is called

Reply:
remove the comment

Reply:
I was reading the code in the IDE and I came here to add the same question 😆 . +1 to remove the comment